seisbench/generate/augmentation.py

```diff
@@ -670,3 +672,194 @@ def __call__(self, state_dict):
+                  If alpha is zero, a new angle is defined each time the method is called, otherwise the angle is
+                  constant.
+                  Default value is None, i.e. the angle changes with every call.
+    :type alpha: None, float, int
+    :param components: Defines which components are rotated.
+                       Default components are N and E.
+    :type components: str
+    :param key: The keys for reading from and writing to the state dict.
+                If key is a single string, the corresponding entry in state dict is modified.
+                Otherwise, a 2-tuple is expected, with the first string indicating the key to
+                read from and the second one the key to write to.
+    :type key: str, tuple[str, str]
+    """
+
+    def __init__(
+        self, key: str = "X", alpha: (None, float, int) = None, components: str = "NE"
+    ):
+        if isinstance(key, str):
+            self.key = (key, key)
+        else:
+            self.key = key
+
+        self.components = components
+        self.alpha = alpha
+
+        if len(self.components) != 2:
+            msg = f"Only two components can be rotated and not {len(self.components)}."
+            raise ValueError(msg)
+
+    def __call__(self, state_dict) -> np.array:
+        x, metadata = state_dict[self.key[0]]
+
+        if self.key[0] != self.key[1]:
+            # Ensure metadata is not modified inplace unless input and output key are anyhow identical
+            metadata = copy.deepcopy(metadata)
+
+        # Create random angle alpha if alpha is not given in init
+        if self.alpha is None:
+            alpha = np.random.uniform(-np.pi, np.pi)
+        else:
+            alpha = self.alpha
+
+        trace_components = metadata.get("trace_component_order")
+        if not trace_components:
+            msg = "Keyword 'trace_component_order' is missing in metadata, therefore rotation is not possible."
+            raise ValueError(msg)
+
+        data_comp1 = copy.copy(x[trace_components.index(self.components[0]), :])
+        data_comp2 = copy.copy(x[trace_components.index(self.components[1]), :])
+
+        # Rotate by angle alpha
+        x[trace_components.index(self.components[0]), :] = data_comp1 * np.cos(
+            alpha
+        ) - data_comp2 * np.sin(alpha)
+        x[trace_components.index(self.components[1]), :] = data_comp1 * np.sin(
+            alpha
+        ) + data_comp2 * np.cos(alpha)
+
+        state_dict[self.key[1]] = (x, metadata)
+
+    def __str__(self):
+        if not self.alpha:
+            return f"Rotation of {self.components} components by arbitrary angle"
+        else:
+            return f"Rotation of {self.components} components by {np.round(self.alpha, 2)} rad"
+
+
+class RealNoise:
+    """
+    Adds recorded noise from an extra noise data set to the data.
+    This method was inspired by the following publication:
+    Jun Zhu, Lihua Fang, Fajun Miao, Liping Fan, Ji Zhang, Zefeng Li
```

Comment:
Could you reformat this into a citation block? Something [like this](https://github.com/seisbench/seisbench/blob/cc180f7824b40d1fb0155e39bbf901549b709115/docs/pages/benchmark_datasets.rst?plain=1#L48).